src/cursor/cur_std.c

```diff
@@ -1113,6 +1113,43 @@ __wt_cursor_reconfigure(WT_CURSOR *cursor, const char *config)
     API_END_RET(session, ret);
 }
 
+/*
+ * __wt_cursor_largest_key --
+ *     WT_CURSOR->largest_key default implementation..
+ */
+int
+__wt_cursor_largest_key(WT_CURSOR *cursor)
+{
+    WT_DECL_RET;
+    WT_ITEM *key;
+    WT_SESSION_IMPL *session;
+
+    CURSOR_API_CALL(cursor, session, largest_key, NULL);
+    WT_ERR(__wt_scr_alloc(session, 0, &key));
+
+    /* Reset the cursor to give up the cursor position. */
+    WT_ERR(cursor->reset(cursor));
+    /* Ignore deletion */
+    F_SET(cursor, WT_CURSTD_IGNORE_TOMBSTONE);
+
+    /* Call cursor prev with read uncommitted isolation level. */
+    WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_UNCOMMITTED, ret = cursor->prev(cursor));
+    WT_ERR(ret);
+
+    /* Copy the key as we will reset the cursor after that. */
+    WT_ERR(__wt_buf_set(session, key, cursor->key.data, cursor->key.size));
```

Comment:
I think it's hard to skip globally visible tombstone with a general implementation for all the cursor types as I think we can't convert all the cursor types to btree cursor and get the time window information.

In addition, I'm not sure is it possible here to get the key as binary format using the get_key interface regardless of the cursor key_format. If that is possible, history store cursor can use the same implementation, otherwise, it is not supported currently.